TestProject/ComplexTests.cs.expected

```diff
@@ -0,0 +1,165 @@
+
+namespace TestProject
+{
+  [TestFixture]
+  public class ComplexTests
+  {
+    private ITestInterface _memberCaller;
+    private Order _order1;
+
+    private Customer _customer1;
+    private DomainObjectMockEventReceiver _order1MockEventReceiver;
+    private DomainObjectMockEventReceiver _customer1MockEventReceiver;
+    private ClientTransaction _subTransaction;
+    private Mock<ITestInterface> _serviceStub;
+    private Mock<ITypeInformation> _typeInformationForResourceResolutionStub;
+    private Mock<ITypeInformation> _typeInformationStub;
+    private Mock<IPropertyInformation> _propertyInformationStub;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _memberCaller = new TestClass();
+      _serviceStub = new Mock<ITestInterface>();
+      _typeInformationStub = new Mock<ITypeInformation>();
+      _typeInformationForResourceResolutionStub = new Mock<ITypeInformation>();
+      _propertyInformationStub = new Mock<IPropertyInformation>();
+    }
+
+    [Test]
+    public void InitRecursive ()
+    {
+      Page namingContainer = new Page();
+      var parentControlMock = new Mock<IControl>() { CallBase = true };
+      var childControlMock = new Mock<IControl>() { CallBase = true };
+      var sequence1 = new MockSequence();
+      childControlMock.InSequence (sequence1).Setup (mock => PrivateInvoke.InvokeNonPublicMethod (mock, "OnInit", EventArgs.Empty)).Verifiable();
+      parentControlMock.InSequence (sequence1).Setup (mock => PrivateInvoke.InvokeNonPublicMethod (mock, "OnInit", EventArgs.Empty)).Verifiable();
+
+      namingContainer.Controls.Add (parentControlMock.Object);
+      parentControlMock.Object.Controls.Add (childControlMock.Object);
+      _memberCaller.InitRecursive (parentControlMock.Object, namingContainer);
+
+      parentControlMock.Verify();
+
+      childControlMock.Verify();
+    }
+
+    [Test]
+    public void SubCommit_OfDeletedObject_DoesNotRaiseDeletedEvent ()
+    {
+      using (_subTransaction.EnterDiscardingScope())
+      {
+        ITestInterface domainObject = new TestClass();
+
+        var extensionMock = new Mock<IClientTransactionExtension> (MockBehavior.Strict);
+        extensionMock.Setup (stub => stub.Key).Returns ("Mock");
+        _subTransaction.Extensions.Add (extensionMock.ToString());
+        try
+        {
+
+          extensionMock.Object.ObjectDeleting (_subTransaction, domainObject);
+          extensionMock.Object.ObjectDeleted (_subTransaction, domainObject);
+          domainObject.Delete();
+          extensionMock.Verify();
+
+          extensionMock.Setup (_ => _.Committing (It.IsAny<ITestInterface>(), It.IsAny<Order>(), It.IsAny<ClientTransaction>())).Verifiable();
+          extensionMock.Setup (_ => _.CommitValidate (It.IsAny<ITestInterface>(), It.IsAny<ClientTransaction>())).Verifiable();
+          extensionMock.Setup (_ => _.Committed (It.IsAny<ITestInterface>(), It.IsAny<ClientTransaction>())).Verifiable();
+          _subTransaction.Commit();
```

Comment:
here is another deleted blank line. 
so, second rule: do not delete the final remaining blank line if the updated/removed code was in the middle of other code.